server/repository/src/migrations/v1_04_00/plugin_data.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{migrations::*, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    #[cfg(not(feature = "postgres"))]
+    const RELATED_RECORD_TYPE: &'static str = "TEXT";
+    #[cfg(feature = "postgres")]
+    const RELATED_RECORD_TYPE: &'static str = "related_record_type";
+    #[cfg(feature = "postgres")]
+    sql!(
+        connection,
+        r#"
+            CREATE TYPE {RELATED_RECORD_TYPE} AS ENUM (
+                'STOCK_LINE'
```

Comment:
Only added the example type for now... Did you want me to add all of them now?

Reply:
Actually, I was thinking, do we need a record type ?

@mark-prins would plugin fetch it's own data ? Or would it be commonly fetched ? Same for the data update ? I think if plugins fetch their own data then we wouldn't need related_record_type ? I can even see a plugin that adds a column to a table, which is queried together with a table (so can be filterable), i.e.: https://www.db-fiddle.com/f/vGbHfXYjP9oWWdq5ajifri/1

select

Reply:
that's a good point - plugins are tied directly to a specific data type, and therefore if the data is only ever fetched or updated by the plugin then the type is irrelevant. 
It does mean that if we are ever running queries in the database directly it would be difficult to include the plugin data - would also potentially cause trouble when trying to query data for reports. That could be difficult anyway, trying to include plugin data in a report!